ludwig/api.py

```diff
@@ -1567,7 +1568,8 @@ def load(
         # Upgrades deprecated fields and adds new required fields in case the config loaded from disk is old.
         config_obj = ModelConfig.from_dict(config)
 
-        if backend_param is None and "backend" in config:
+        # Ensure that the original backend is used if it was specified in the config and user requests it
+        if use_backend_from_config or (backend_param is None and "backend" in config):
```

Comment:
Yeah, so the issue comes from the evaluate CLI: https://github.com/ludwig-ai/ludwig/blob/adc82cc33c041f29480f734bf146a5d02b5e8641/ludwig/evaluate.py#L267-L274

The backend is initialized fresh when running `ludwig evaluate`. In doing this, the backend config is entirely ignored. This made it difficult to iterate quickly on this PR (my strategy was to run `ludwig train` once, then run `ludwig evaluate` to debug batch evaluation and prediction).

This is not a problem in Predibase engine code though, because backend is passed in as None and therefore the config is respected every time.

Reply:
Okay, going to push a quick change to fix the issue here, by not plumbing through the backend used in the CLI evaluate code path.